lib/models/application.coffee

```diff
@@ -237,24 +257,25 @@ getApplicationModel = (deps, opts) ->
 	# @function
 	# @memberof resin.models.application
 	#
-	# @param {String} name - application name
+	# @param {String|Number} nameOrId - application name (string) or id (number)
 	# @returns {Promise}
 	#
 	# @example
 	# resin.models.application.remove('MyApp');
 	#
 	# @example
+	# resin.models.application.remove(123);
+	#
+	# @example
 	# resin.models.application.remove('MyApp', function(error) {
 	# 	if (error) throw error;
 	# });
 	###
-	exports.remove = (name, callback) ->
-		exports.get(name).then ->
+	exports.remove = (nameOrId, callback) ->
+		exports.get(nameOrId).then (application) ->
```

Comment:
Oh sorry skipped that. Unless Juan knows a good reason — good to remove I think.
Different errors for not found are probably OK.

But there's a separate question about the multiple found entries.
Both app names and device UUIDs are unique for the given user. But the user can have access (and thus see) two apps of the same name if one of them is shared with the user. What will the SDK do in this case? According to the code (`.get(0)`) it will just randomly pick the first of them which sounds like an issue to me.